pkg/controller/elasticsearch/driver/upgrade_predicates.go

```diff
@@ -126,21 +135,50 @@ var predicates = [...]Predicate{
 			deletedPods []corev1.Pod,
 			maxUnavailableReached bool,
 		) (b bool, e error) {
-			// If candidate is not a master then we don't care
+
+			// If candidate is not a master then we just check if it will become a master
+			// In this case we account for a master creation as we want to avoid creating more
+			// than one master at a time.
 			if !label.IsMasterNode(candidate) {
+				if stringsutil.StringInSlice(candidate.Name, context.masterNodesNames) {
```

Comment:
Maybe worth putting this behind a speaking function name like `func willBecomeMasterNode` or something like that